qiskit/compiler/transpile.py

```diff
@@ -120,6 +120,36 @@ def transpile(circuits,
             pass manager will be used directly (Qiskit will not attempt to
             auto-select a pass manager based on transpile options).
 
+        callback (func):
+            A callback function that will be called after each
+            pass execution. The function will be called with 5 keyword
+            arguments:
+                pass_ (Pass): the pass being run
+                dag (DAGCircuit): the dag output of the pass
+                time (float): the time to execute the pass
+                property_set (PropertySet): the property set
+                count (int): the index for the pass execution
+
+            Keyword arguments are used for forward compatibility reasons,
+            and the current options will always be available moving
+            forward. But, additional arguments may be added in future
+            release. To ensure your callback functions work against any
+            version of terra moving forward you want to have your callback
+            function take a single `**kwargs` input.
```

Comment:
I'm not sure backwards compatibility is so important here, and something we should promise at this stage. This is just a debugging tool that should be used sporadically, not something to write notebooks with, etc.
For example if we merge DAGCircuit with QuantumCircuit, then the `dag` kwarg will be lost.

Reply:
Sure, I'll reword this to instead warn there is no interface guarantee moving forward.